packages/react/src/components/select/custom/useDropdownKeyboardKey.spec.ts

```diff
@@ -0,0 +1,539 @@
+import { describe, expect, it } from '@jest/globals';
+import { Dispatch, MutableRefObject, SetStateAction } from 'react';
+import { IndexedOption, Option } from './useCustomSelect';
+import { useDropdownKeyboardKey } from './useDropdownKeyboardKey';
+
+let callbackFunc: (
+  key: React.KeyboardEvent<Element>['key'],
+  haltEvent: () => void
+) => void;
+
+jest.mock('../../../hooks', () => ({
+  useKeyboardKey: jest
+    .fn()
+    .mockImplementation((_, callback: any) => (callbackFunc = callback)),
+}));
+
+describe('useDropdownKeyboardKey', () => {
```

Comment:
Don't know even what to think about this test, over 500 lines... 😟

Reply:
I know right, which is why I'm always wary of introducing JS to do something the DOM can natively.

As I said in the other comment, we can iterate and refactor this component without API changes in the near future, hopefully cutting a lot of the custom code.